rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -225,4 +228,49 @@ public static boolean isValidJSONObject(String str) {
             return false;
         }
     }
+
+    public static void verifyUserAuthorization() {
+        SyncUtils syncUtils = new SyncUtils(RDTApplication.getInstance());
+        UserAuthorizationVerificationTask userAuthorizationVerificationTask = UserAuthorizationVerificationTask.getInstance(syncUtils);
+        if (userAuthorizationVerificationTask.getStatus().equals(AsyncTask.Status.PENDING)) {
+            userAuthorizationVerificationTask.execute();
```

Comment:
We won't need this block after the changes

Reply:
I think we need this block because the first-time task state is PENDING.

Reply:
Refactor to this:

```
        if   (userAuthorizationVerificationTask.getStatus().equals(AsyncTask.Status.FINISHED)) {
              userAuthorizationVerificationTask.destroyInstance();
        }
        UserAuthorizationVerificationTask.getInstance(context).execute();
```

Reply:
let's consider the case of a user currently on the home screen then the task will execute and then the user moves to profile activity and somehow previous task is not finish i.e task state is RUNNING so this task will run again instead of skipping.

Reply:
I don't think so, the first check is that the task is not running. Either way, a unit test can verify this.

Reply:
fixed.